File: 1009-pancake-sorting/pancake-sorting.cpp
class Solution {
public:
    vector<int> pancakeSort(vector<int>& arr) {
        vector<int> res;
        int n = arr.size();
        for(int currsize = n; currsize>1;currsize--){
            int idx = 0;
            for(int i = 1; i<currsize; i++){
                if(arr[i]>arr[idx]){
                    idx = i;
                }
            }
            if(idx == currsize){
                continue;
            }
            if(idx!=0){
                res.push_back(idx+1);
                reverse(arr.begin(),arr.begin()+idx+1);
            }
            res.push_back(currsize);
            reverse(arr.begin(),arr.begin()+currsize);
        }
        return res;
    }
};

File: 1009-pancake-sorting/pancake-sorting.py
class Solution:
    def pancakeSort(self, arr: List[int]) -> List[int]:
        res = []
        n = len(arr)
        for curr in range(n,1,-1):
            idx = 0
            for i in range(1,curr):
                if(arr[i]> arr[idx]):
                    idx = i
            if(idx == curr - 1):
                continue
            if(idx!=0):
                res.append(idx+1)
                arr[:idx + 1] = arr[:idx +1][::-1]

            res.append(curr)
            arr[:curr] = arr[:curr][::-1]
        return res